ext/standard/tests/general_functions/get_debug_type_basic.phpt

```diff
@@ -0,0 +1,43 @@
+--TEST--
+Test get_debug_type() class reading
+--FILE--
+<?php
+
+namespace Demo {
+    class ClassInNamespace {
+
+    }
+}
+
+namespace {
+    class ClassInGlobal {
+
+    }
+
+    /* tests against an object type */
+    echo get_debug_type(new ClassInGlobal()) . "\n";
+    echo get_debug_type(new Demo\ClassInNamespace()) . "\n";
+
+    /* scalars */
+    echo get_debug_type("foo") . "\n";
+    echo get_debug_type(false) . "\n";
+    echo get_debug_type(true) . "\n";
+    echo get_debug_type(1) . "\n";
+    echo get_debug_type(1.1) . "\n";
+    echo get_debug_type([]) . "\n";
+    echo get_debug_type(null) . "\n";
+    echo get_debug_type(fopen(__FILE__, 'r')) . "\n";
+
+}
+
+--EXPECT--
+ClassInGlobal
+Demo\ClassInNamespace
+string
+bool
+bool
+int
```

Comment:
Not sure about using in-code names.
Error messages are more readable when a sentence reads as "integer given".

Reply:
It's very important to use the canonical type names here, everything else causes confusion for users. We spent a lot of effort to eradicate uses of `boolean`, `integer` and `double` everywhere apart from `gettype()`. Please ensure that you do not use these in Symfony either.

Reply:
We use `is_object($v) ? get_class($v) : gettype($v)` a lot so we use these in Symfony for sure (and we won't change as that'd mean more boilerplate - but that's not an objection against your point).